### PR TITLE
Generator creation dialog

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/components/dialogs/generator-creation-dialog.js [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+                            <FormControlLabel
+                                className={classes.voltageRegulation}
+                                id="voltage-regulation"
+                                control={
+                                    <Switch
+                                        checked={voltageRegulation}
+                                        onChange={(e) =>
+                                            handleChangeVoltageRegulation(e)
+                                        }
+                                        value="checked"
```
Comment (sBouzols) [inline src/components/dialogs/generator-creation-dialog.js:587]: What is the purpose of value for a Switch component?

File: src/components/dialogs/generator-creation-dialog.js [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+                                    />
+                                }
+                                label={intl.formatMessage({
+                                    id: 'VoltageRegulationText',
+                                })}
+                            />
+                        </Grid>
+                        <Grid item xs={4} align="left">
+                            <TextFieldWithAdornment
+                                id="voltage-set-point"
```
Comment (sBouzols) [inline src/components/dialogs/generator-creation-dialog.js:601]: fullWidth + variant="outlined"

File: src/components/dialogs/generator-creation-dialog.js [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+                    <br />
+
+                    {/* Connectivity part */}
+                    <ConnectivityDialog
+                        network={network}
+                        voltageLevel={voltageLevel}
+                        busOrBusbarSection={busOrBusbarSection}
+                        busOrBusbarSectionOptions={busOrBusbarSectionOptions}
+                        errors={errors}
+                        changeBusOrBusbarSectionOptions={(
```
Comment (sBouzols) [inline src/components/dialogs/generator-creation-dialog.js:631]: remove

File: src/components/util/validation-functions.js [hunk 4]
```diff
@@ -26,6 +26,10 @@ export function validateField(value, toValidate) {
         }
     }
 
+    if (toValidate.checkFieldValue) {
+        return toValidate.checkFieldValue();
```
Comment (sBouzols) [inline src/components/util/validation-functions.js:30]: Why not checking positivity or interval belonging in here ?

File: src/components/dialogs/connectivity-dialog.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+        margin: 0,
+    },
+    popper: {
+        style: {
+            width: 'fit-content',
+        },
+    },
+}));
+
+const ConnectivityDialog = ({
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/connectivity-dialog.js]: done

File: src/components/dialogs/generator-creation-dialog.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:587]: property removed

File: src/components/dialogs/load-creation-dialog.js [hunk 6]
```diff
@@ -29,17 +22,10 @@ import {
...
 import TextFieldWithAdornment from '../util/text-field-with-adornment';
 import { validateField } from '../util/validation-functions';
-
-// Factory used to create a filter method that is used to change the default
-// option filter behaviour of the Autocomplete component
-const filter = createFilterOptions();
+import ConnectivityDialog from './connectivity-dialog';
 
 const useStyles = makeStyles((theme) => ({
     helperText: {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/load-creation-dialog.js:31]: removed

File: src/components/dialogs/generator-creation-dialog.js [hunk 7]
```diff
@@ -0,0 +1,663 @@
...
+import { validateField } from '../util/validation-functions';
+import { makeStyles } from '@material-ui/core/styles';
+import ConnectivityDialog from './connectivity-dialog';
+
+const useStyles = makeStyles((theme) => ({
+    helperText: {
+        margin: 0,
+        marginTop: 3,
+    },
+    popper: {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:37]: removed

File: src/components/dialogs/connectivity-dialog.js [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    network,
+    voltageLevel,
+    busOrBusbarSection,
+    busOrBusbarSectionOptions,
+    errors,
+    changeVoltageLevel,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/connectivity-dialog.js]: renamed

File: src/components/dialogs/generator-creation-dialog.js [hunk 9]
```diff
@@ -0,0 +1,663 @@
...
+        style: {
+            width: 'fit-content',
+        },
+    },
+    dialogPaper: {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:42]: styles removed

File: src/components/dialogs/generator-creation-dialog.js [hunk 10]
```diff
@@ -0,0 +1,663 @@
...
+    const [voltageSetpoint, setVoltageSetpoint] = useState('');
+    const [enabledVoltageSetpoint, setEnabledVoltageSetpoint] = useState(false);
+
+    const [voltageLevel, setVoltageLevel] = useState(null);
+
+    const [busOrBusbarSection, setBusOrBusbarSection] = useState(null);
+
+    const [errors, setErrors] = useState(new Map());
+
+    const handleChangeGenertorId = (event) => {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:103]: corrected

File: src/components/dialogs/generator-creation-dialog.js [hunk 11]
```diff
@@ -0,0 +1,663 @@
...
+        // Check if error list contains an error
+        let isValid =
+            Array.from(tmpErrors.values()).findIndex((err) => err.error) === -1;
+
+        if (isValid) {
+            createGenerator(
+                studyUuid,
+                generatorId,
+                generatorName ? generatorName : null,
+                !energySource ? 'OTHER' : energySource,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:313]: this field is optional, and if not given by the user, the energy source is set to 'OTHER'

File: src/components/dialogs/generator-creation-dialog.js [hunk 12]
```diff
@@ -0,0 +1,663 @@
...
+    const classes = useStyles();
+
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const intl = useIntl();
+    const intlRef = useIntlRef();
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [busOrBusbarSectionOptions, setBusOrBusbarSectionOptions] = useState(
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:70]: removed

File: src/components/dialogs/generator-creation-dialog.js [hunk 13]
```diff
@@ -0,0 +1,663 @@
...
+        setErrors(tmpErrors);
+    };
+
+    const handleChangeVoltageSetpoint = (event) => {
+        // TODO: remove replace when parsing behaviour will be made according to locale
+        // Replace ',' by '.' to ensure double values can be parsed correctly
+        setVoltageSetpoint(event.target.value?.replace(',', '.'));
+    };
+
+    const checkMinMaxPower = (minPower, maxPower) => {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js:156]: Done

**Review by niconoir — APPROVED** [review]

**Pull request merged**